runtime/src/iree/hal/drivers/vulkan/util/intrusive_list.h

```diff
@@ -204,10 +204,26 @@ class IntrusiveListBase {
   size_t count_ = 0;
 };
 
+namespace impl {
+
+// std::iterator has been deprecated in C++17, therefore we create our own
+// definition of the required types for iterators.
+template <typename Category, typename T, typename Distance = std::ptrdiff_t,
+          typename Pointer = T*, typename Reference = T&>
+struct iterator {
```

Comment:
Perhaps Iterator is a better name, I used iterator to keep it consistent with std::iterator, and the Lint didn't flag it.